build.gradle

```diff
@@ -31,23 +28,36 @@ processResources {
 	}
 }
 
-// ensure that the encoding is set to UTF-8, no matter what the system default is
-// this fixes some edge cases with special characters not displaying correctly
-// see http://yodaconditions.net/blog/fix-for-java-file-encoding-problems-with-gradle.html
-tasks.withType(JavaCompile) {
-	options.encoding = "UTF-8"
+tasks.withType(JavaCompile).configureEach {
+	// ensure that the encoding is set to UTF-8, no matter what the system default is
+	// this fixes some edge cases with special characters not displaying correctly
+	// see http://yodaconditions.net/blog/fix-for-java-file-encoding-problems-with-gradle.html
+	// If Javadoc is generated, this must be specified in that task too.
+	it.options.encoding = "UTF-8"
+
+	// The Minecraft launcher currently installs Java 8 for users, so your mod probably wants to target Java 8 too
+	// JDK 9 introduced a new way of specifying this that will make sure no newer classes or methods are used.
+	// We'll use that if it's available, but otherwise we'll use the older option.
+	def targetVersion = JavaVersion.VERSION_1_8
+	if (JavaVersion.current().isJava9Compatible()) {
+		 it.options.release = targetVersion.ordinal() + 1
+	} else {
+		 it.sourceCompatibility = targetVersion
+		 it.targetCompatibility = targetVersion
+	}
 }
 
-// Loom will automatically attach sourcesJar to a RemapSourcesJar task and to the "build" task
-// if it is present.
-// If you remove this task, sources will not be generated.
-task sourcesJar(type: Jar, dependsOn: classes) {
-	classifier = "sources"
-	from sourceSets.main.allSource
+java {
+	// Loom will automatically attach sourcesJar to a RemapSourcesJar task and to the "build" task
+	// if it is present.
+	// If you remove this line, sources will not be generated.
+	withSourcesJar()
 }
 
 jar {
-	from "LICENSE"
+	from("LICENSE") {
+		rename { "${it}_${project.name}"}
```

Comment:
This is not that important, but this can cause problems if a project (most likely a library in this case) doesn't set the project name in gradle.properties. If it's built using Jitpack (for example for jij'ing), the file will be `LICENSE_build` as Gradle guesses the project name based on the directory, which Jitpack calls `build`.

Reply:
ah -- I usually set the project name in `settings.gradle`. I might update this PR to skip setting `archivesBaseName` and instead just set the project name explicitly.